### PR TITLE
beyond-compare@4 4.4.7.28397 (new cask)

### PR DESCRIPTION
**Pull request by jonpham**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

This file is a copy of last year's version of this cask 
https://github.com/Homebrew/homebrew-cask/blob/304ba4bce8d3c460fba93df2e45e7f59ef1c88ab/Casks/b/beyond-compare.rb

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/b/beyond-compare@4.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  version "4.4.7.28397"
+  sha256 "a9ba4cea125bbfe00fa3e79de39937197ae5d479d94710f5b93da5bda377a0ce"
+
+  url "https://www.scootersoftware.com/BCompareOSX-#{version}.zip"
+  name "Beyond Compare"
+  desc "Compare files and folders"
+  homepage "https://www.scootersoftware.com/"
+
+  livecheck do
+    url "https://www.scootersoftware.com/download.php?zz=v4changelog"
```
Comment (khipp) [inline Casks/b/beyond-compare@4.rb:11]:
```suggestion
    url "https://www.scootersoftware.com/download/v4"
```

File: Casks/b/beyond-compare@4.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    regex(/BCompareOSX[_.-]v?(\d+(?:\.\d+)+)\.zip/i)
+  end
+
+  auto_updates true
```
Comment (khipp) [inline Casks/b/beyond-compare@4.rb:15]:
```suggestion
  auto_updates true
  conflicts_with cask: [
    "beyond-compare",
    "beyond-compare@beta",
  ]
```

File: Casks/b/beyond-compare@4.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  app "Beyond Compare.app"
+  binary "#{appdir}/Beyond Compare.app/Contents/MacOS/bcomp"
+
+  zap trash: [
+    "~/Library/Application Support/Beyond Compare",
+    "~/Library/Caches/com.apple.helpd/Generated/Beyond Compare Help*",
+    "~/Library/Caches/com.apple.helpd/Generated/com.ScooterSoftware.BeyondCompare.help*",
+    "~/Library/Caches/com.ScooterSoftware.BeyondCompare",
+    "~/Library/Saved Application State/com.ScooterSoftware.BeyondCompare.savedState",
```
Comment (khipp) [inline Casks/b/beyond-compare@4.rb:25]:
```suggestion
    "~/Library/Application Support/Beyond Compare*",
    "~/Library/Caches/com.apple.helpd/Generated/Beyond Compare Help*",
    "~/Library/Caches/com.apple.helpd/Generated/com.ScooterSoftware.BeyondCompare.help*",
    "~/Library/Caches/com.ScooterSoftware.BeyondCompare",
    "~/Library/Containers/com.ScooterSoftware.BeyondCompare.BCFinder",
    "~/Library/Preferences/com.ScooterSoftware.BeyondCompare.plist",
    "~/Library/Saved Application State/com.ScooterSoftware.BeyondCompare.savedState",
```

File: Casks/b/beyond-compare@4.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  ]
+end
```
Comment (khipp) [inline Casks/b/beyond-compare@4.rb:27]:
```suggestion

  caveats do
    requires_rosetta
  end
end

```

**Review by khipp — COMMENTED** [review]
Thank you for your contribution, @jonpham! I left a few suggestions below.

File: Casks/b/beyond-compare@4.rb (hunk 1 above)
Comment (krehel) [reply Casks/b/beyond-compare@4.rb:11]: Do we need a livecheck here?  Does this version receive updates?  Looks like the last update was over a year ago.

Comment (bevanjkay) [reply Casks/b/beyond-compare@4.rb:11]: I don't think there's any harm in including a `livecheck` wherever they remain functional.

Comment (jonpham) [reply Casks/b/beyond-compare@4.rb]: Left it in there, but also changed the download link to have a more rigorous regex for v4

File: Casks/b/beyond-compare@4.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  sha256 "a9ba4cea125bbfe00fa3e79de39937197ae5d479d94710f5b93da5bda377a0ce"
+
+  url "https://www.scootersoftware.com/BCompareOSX-#{version}.zip"
+  name "Beyond Compare"
+  desc "Compare files and folders"
+  homepage "https://www.scootersoftware.com/"
+
+  livecheck do
+    url "https://www.scootersoftware.com/download/v4"
+    regex(/BCompareOSX[_.-]v?(4\.{1}\d+(?:\.\d+)+)\.zip/i)
```
Comment (khipp) [inline Casks/b/beyond-compare@4.rb:12]:
```suggestion
    regex(/BCompareOSX[._-]v?(\d+(?:\.\d+)+)\.zip/i)
```

We should be able to stick to our usual patterns and keep this as it is, as we have already fixed the `livecheck` URL to v4.

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @jonpham! 🎉

**Review by p-linnane — APPROVED** [review]

**Pull request merged**